atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/progress/SweepProgressStore.java

```diff
@@ -61,17 +79,34 @@ public void clearProgress() {
         // 1) The table should be small, performance difference should be negligible.
         // 2) Truncate takes an exclusive lock in Postgres, which can interfere
         // with concurrently running backups.
-        kvs.deleteRange(tableFactory.getSweepProgressTable(null).getTableRef(), RangeRequest.all());
+        kvs.deleteRange(tableRef, RangeRequest.all());
     }
 
-    private static SweepProgress hydrateProgress(SweepProgressTable.SweepProgressRowResult rr) {
-        return ImmutableSweepProgress.builder()
-                .tableRef(TableReference.createUnsafe(rr.getFullTableName()))
-                .startRow(rr.getStartRow())
-                .cellTsPairsExamined(rr.getCellsExamined())
-                .staleValuesDeleted(rr.getCellsDeleted())
-                .minimumSweptTimestamp(rr.getMinimumSweptTimestamp())
-                .build();
+    private CheckAndSetRequest casProgressRequest(Optional<SweepProgress> oldProgress, SweepProgress newProgress)
+            throws JsonProcessingException {
+        if (!oldProgress.isPresent()) {
+            return CheckAndSetRequest.newCell(tableRef, CELL, progressToBytes(newProgress));
+        }
+        return CheckAndSetRequest.singleCell(tableRef,
+                CELL, progressToBytes(oldProgress.get()), progressToBytes(newProgress));
     }
 
+    private byte[] progressToBytes(SweepProgress value) throws JsonProcessingException {
+        return OBJECT_MAPPER.writeValueAsBytes(value);
+    }
+
+    public static Optional<SweepProgress> hydrateProgress(Map<Cell, Value> result) {
```

Comment:
Should this be `private`? Or do we have cases where we might want to hydrate the progress externally?